pkg/dnsmasq/nanny.go

```diff
@@ -68,7 +70,30 @@ func (n *Nanny) Configure(args []string, config *config.Config) {
 	}
 
 	for domain, serverList := range config.StubDomains {
+		resolver := &net.Resolver{
+			PreferGo: true,
+			Dial: func(ctx context.Context, network, address string) (net.Conn, error) {
+				d := net.Dialer{}
+				return d.DialContext(ctx, "udp", kubednsServer)
+			},
+		}
 		for _, server := range serverList {
+			if isIP := (net.ParseIP(server) != nil); !isIP {
+				switch {
+				case strings.HasSuffix(server, "cluster.local"):
+					if IPs, err := resolver.LookupIPAddr(context.Background(), server); err != nil {
+						glog.Errorf("Error looking up IP for %s: %v", server, err)
+					} else if len(IPs) > 0 {
```

Comment:
if len(IPs) == 0 {
  glog.Errorf("Name %q does not resolve to any IPs", server)
}

add to the default as well.